flink-runtime/pom.xml

```diff
@@ -402,6 +402,59 @@ under the License.
 				</configuration>
 			</plugin>
 
+			<plugin>
+				<groupId>com.google.code.maven-replacer-plugin</groupId>
```

Comment:
To me this is the cleanest way to get the correct values needed.

Reply:
cc @zentol

Reply:
We synced offline with @zentol. In general, we agree that this is a cleaner way to go as we explicitly use versions defined in one place. Nonetheless, as I mentioned, this is a separate issue which could be a prerequisite for the change at hand (default value of Flink image for native k8s integration). There are some open questions to resolve in that separate issue.  E.g. the setup of most IDE projects  will be broken w/o code generation as `Version` class is not committed, so this is going to be annoying. If we commit it, it can easily become stale then it is better to keep then "unknown" for IDE builds and generate into the actual built artefacts. Committing the actual values all the time is also annoying, especially for the git data.

Reply:
OK, I'll create a separate issue for generating the Versions file. Note that in my experience (IntelliJ) has a generate sources button in the area of maven which simply regenerates (among other things) this file. This is exactly the same pattern as used with all other generated code; for example Antlr4.

Reply:
I created https://issues.apache.org/jira/browse/FLINK-16871 . Would you please assign this to me? Thanks.